WhisperAPI/WhisperAPI.Tests/Unit/ModelValidationTest.cs

```diff
@@ -0,0 +1,116 @@
+            this.ValidateModel(query).Should().BeFalse();
+        }
+
+        [Test]
+        public void When_max_documents_is_positive_return_model_state_valid()
+        {
+            var query = SuggestionQueryBuilder.Build.WithChatKey(new Guid("0f8fad5b-d9cb-469f-a165-708677289502")).WithMaxDocuments(5).Instance;
+            this.ValidateModel(query).Should().BeTrue();
+        }
+
+        [Test]
+        public void When_max_questions_is_negative_return_model_state_invalid()
+        {
+            var query = SuggestionQueryBuilder.Build.WithChatKey(new Guid("0f8fad5b-d9cb-469f-a165-708677289502")).WithMaxQuestions(-2).Instance;
+            this.ValidateModel(query).Should().BeFalse();
+        }
+
+        [Test]
+        public void When_max_questions_is_positive_return_model_state_valid()
+        {
+            var query = SuggestionQueryBuilder.Build.WithChatKey(new Guid("0f8fad5b-d9cb-469f-a165-708677289502")).WithMaxQuestions(5).Instance;
+            this.ValidateModel(query).Should().BeTrue();
+        }
+
+        [Test]
+        [TestCase(5, -1)]
+        [TestCase(-5, 1)]
+        [TestCase(-5, -1)]
+        public void When_max_questions_or_max_documents_is_negative_return_model_state_invalid(int maxDocuments, int maxQuestions)
+        {
+            var query = SuggestionQueryBuilder.Build.WithChatKey(new Guid("0f8fad5b-d9cb-469f-a165-708677289502")).WithMaxDocuments(maxDocuments).WithMaxQuestions(maxQuestions).Instance;
+            this.ValidateModel(query).Should().BeFalse();
+        }
+
+        [Test]
+        [TestCase(5, 1)]
+        [TestCase(0, 1)]
+        [TestCase(5, 0)]
+        public void When_max_questions_and_max_documents_are_positive_return_model_state_valid(int maxDocuments, int maxQuestions)
+        {
+            var query = SuggestionQueryBuilder.Build.WithChatKey(new Guid("0f8fad5b-d9cb-469f-a165-708677289502")).WithMaxDocuments(maxDocuments).WithMaxQuestions(maxQuestions).Instance;
+            this.ValidateModel(query).Should().BeTrue();
+        }
+
+        private bool ValidateModel(SuggestionQuery query)
+        {
+            var actionContext = new ActionContext(
+                new Mock<HttpContext>().Object,
+                new Mock<RouteData>().Object,
+                new Mock<ActionDescriptor>().Object);
+
+            var actionExecutingContext = new ActionExecutingContext(
+                actionContext,
+                new List<IFilterMetadata>(),
+                new Dictionary<string, object>(),
+                this._contexyController);
+            actionExecutingContext.ActionArguments["query"] = query;
+            if (query != null)
+            {
+                var context = new ValidationContext(query, null, null);
+                var results = new List<ValidationResult>();
+
+                if (!Validator.TryValidateObject(query, context, results, true))
```

Comment:
Can be refactored, we don't need the result. We simply want to know if there's a error or not. can directly return the value of Validator.TryValidateObject(...)